### PR TITLE
feat: make permit2 compatible with gnosis

### PR DESCRIPTION
**Pull request by rndquu**
resolves https://github.com/ubiquity/ubiquibot/issues/285

### CODE REVIEW

File: src/configs/shared.ts [hunk 1]
```diff
@@ -25,7 +25,6 @@ export const DEFAULT_DISQUALIFY_TIME = "7 days"; // 7 days
 export const DEFAULT_CHAIN_ID = 1; // ethereum
 export const DEFAULT_RPC_ENDPOINT = "https://rpc-bot.ubq.fi/v1/mainnet";
 
-export const DEFAULT_PAYMENT_TOKEN = "0x6B175474E89094C44Da98b954EedeAC495271d0F";
```
Comment (pavlovcik) [inline src/configs/shared.ts:28]: Shouldn't we replace this with another default instead of removing?

File: src/helpers/payout.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * 1. Gelato network should support the added chain (https://docs.gelato.network/developer-services/relay/api#oracles)
+ * 2. Gelato network should support the added payment token (https://docs.gelato.network/developer-services/relay/api#oracles-chainid-paymenttokens)
+ */
+
+import { Static } from "@sinclair/typebox";
+import { DEFAULT_RPC_ENDPOINT } from "../configs";
+import { PayoutConfigSchema } from "../types";
+
+// chain ids
+const CHAIN_GNOSIS = 100;
```
Comment (pavlovcik) [inline src/helpers/payout.ts:19]: I just realized that the interesting chains to integrate with for me personally:

Mainnet: 1
Optimism: 10
Gnosis Chain: 100

File: src/configs/shared.ts (hunk 1 above)
Comment (rndquu) [reply src/configs/shared.ts:28]: We don't need the `DEFAULT_PAYMENT_TOKEN` variable anymore so it was deleted. Now we have a separate payment token for [each chain](https://github.com/ubiquity/ubiquibot/pull/316/files#diff-7a573dc385193e6a51c6e9a66e89d361cb981e4c10d27e77d398aa669d9f980dR22).

File: src/helpers/payout.ts (hunk 2 above)
Comment (rndquu) [reply src/helpers/payout.ts:19]: Adding a new chain is basically 6 lines of code:
```js
const CHAIN_OPTIMISM = 10;
const PAYMENT_TOKEN_CHAIN_OPTIMISM = "0x00"; // some token supported by gelato
if (chainId === CHAIN_OPTIMISM) {
    payoutConfig.rpc = "https://rpc.optimism.com";
    payoutConfig.paymentToken = PAYMENT_TOKEN_CHAIN_OPTIMISP;
}
```

Then a partner would need to:
1. Update `chain-id` in the config
2. [Approve permit2 contract to spend tokens](https://dao.ubq.fi/ubiquibot-setup#42f485283d0d463c8a2c8289ecfb8248)

**Review by pavlovcik — APPROVED** [review]

**Comment by 0xcodercrane** [comment]
Here's another thing I am thinking for the gnosis chain. so basically users want to DAI instead of WXDAI on gnosis because it is the native token. Its not a blocker but we might need to think about that on users end

**Comment by rndquu** [comment]
> Here's another thing I am thinking for the gnosis chain. so basically users want to DAI instead of WXDAI on gnosis because it is the native token. Its not a blocker but we might need to think about that on users end

There is no DAI in gnosis chain per se. When you bridge DAI from ethereum mainnet to gnosis chain those bridged DAI are converted to XDAI (`ERC20 token DAI in mainnet = gas token XDAI in gnosis`) which is a native token in gnosis chain used for gas payments. So basically supporting XDAI (native gas token) means that we should also support payments in ETH which means 2 codebases supporting payments both in native (gas) tokens and ERC20 tokens.

**Comment by 0xcodercrane** [comment]
It could be done by creating a permit contract with a wrapper feature on gnosis. its simple but yeah we can do it later

**Comment by pavlovcik** [comment]
xDAI isn't an ERC20

**Comment by 0xcodercrane** [comment]
> xDAI isn't an ERC20

Right but WXDAI is an ERC20 lol

**Comment by pavlovcik** [comment]
Yes, I know. There's some miscommunication but I'll delegate my vote to rndquu because I believe that they did quite a bit of research on the topic.

**Pull request merged**